docs/contribute.rst

```diff
@@ -134,6 +146,57 @@ Find the resulting html pages under the ``docs/_build/html``
 folder.
 
 
+Merging
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Pull request merging is done by maintainers who have a good
+overview of the whole code. Before merging a pull request it's
+good to check the following:
+
+* New test coverage added if appropriate, all tests passed
+* Documentation has been added or updated where appropriate
+* Commit messages are sane, commits are reasonably squashed
+* At least one positive review provided by the maintainers
+* Merge commits are not used, rebase on the master instead
+
+Pull requests which should/cannot not be merged are marked with
+the ``blocked`` label. For complex topics which need more eyes to
+review and discuss before merging use the ``discuss`` label.
```

Comment:
I'd perhaps add (maybe not into merging section but somewhere else) a note about the draft feature on github. Something along the lines of "If your pull request is not completely ready but would like to get some early feedback on the concept, create a draft pull request"

Reply:
One more thing - we discussed on one of the previous hacking sessions that the preferred way of fixing a PR after a review is by adding new commits (to avoid having to go through the whole code again) and rebasing/squashing before merge. This could be something worth noting too.

Reply:
Very good points. Doc updated.